client/components/payment-activity/index.tsx

```diff
@@ -12,22 +20,220 @@ import interpolateComponents from '@automattic/interpolate-components';
 
 import EmptyStateAsset from 'assets/images/payment-activity-empty-state.svg?asset';
 import PaymentActivityData from './payment-activity-data';
+import DateRange from './date-range';
 import Survey from './survey';
 import { WcPayOverviewSurveyContextProvider } from './survey/context';
 import './style.scss';
 
+interface DateRange {
+	start: moment.Moment | undefined;
+	end: moment.Moment | undefined;
+}
+
 const PaymentActivity: React.FC = () => {
 	const { lifetimeTPV } = wcpaySettings;
 	const hasAtLeastOnePayment = lifetimeTPV > 0;
 	const isOverviewSurveySubmitted =
 		wcpaySettings.isOverviewSurveySubmitted ?? false;
 
+	const [ dateRangeState, setDateRangeState ] = useState( {} as DateRange );
+	const [ dateRangePresetState, setDateRangePresetState ] = useState(
+		'last_7_days'
+	);
+	const dateRangePresets = [
+		{
+			value: 'today',
+			label: 'Today',
+		},
+		{
+			value: 'last_7_days',
+			label: 'Last 7 days',
+		},
+		{
+			value: 'last_4_weeks',
+			label: 'Last 4 weeks',
+		},
+		{
+			value: 'last_3_months',
+			label: 'Last 3 months',
+		},
+		{
+			value: 'last_12_months',
+			label: 'Last 12 months',
+		},
+		{
+			value: 'month_to_date',
+			label: 'Month to date',
+		},
+		{
+			value: 'quarter_to_date',
+			label: 'Quarter to date',
+		},
+		{
+			value: 'year_to_date',
+			label: 'Year to date',
+		},
+		{
+			value: 'all_time',
+			label: 'All time',
+		},
+	];
+	const dateRangePresetOnChangeHandler = ( newDateRangePreset: string ) => {
+		let start, end;
+
+		setDateRangePresetState( newDateRangePreset );
+
+		switch ( newDateRangePreset ) {
+			case 'today': {
+				const now = moment();
+				start = now
+					.clone()
+					.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } );
+				end = now.clone().set( {
+					hour: 23,
+					minute: 59,
+					second: 59,
+					millisecond: 0,
+				} );
+				break;
+			}
+			case 'last_7_days': {
+				const now = moment();
+				start = now
+					.clone()
+					.subtract( 7, 'd' )
```

Comment:
This should be 6 instead of 7.

Reply:
Should be 7 and date end should be yesterday, not today.

Discussed in https://github.com/Automattic/woocommerce-payments/pull/8751#discussion_r1608051182